net/ser2net/files/ser2net.init

```diff
@@ -90,90 +100,114 @@ ser2net_led() {
-	local key boolval options
+	local key boolval options custom_options
+	local echo_options=1
 
 	config_get_bool enabled "$cfg" enabled 0
+
+	config_get device "$cfg" device
+	[ -z "$device" ] && return 1
+
 	config_get port "$cfg" port
-	[ "$port" -le 0 -o "$port" -gt 65535 ] && return 1
+	[ "$port" -ge 1 ] && [ "$port" -le 65535 ] || return 1
 
 	config_get protocol "$cfg" protocol
 	case "$protocol" in
-		raw) protocol="tcp" ;;
-		rawlp) protocol="tcp" ; options="$options,wronly" ;;
-		telnet) protocol="telnet,tcp" ;;
-		off) enabled=0 ;;
+		raw)
+			protocol="tcp"
+			;;
+		rawlp)
+			protocol="tcp"
+			options="wronly"
+			;;
+		telnet)
+			protocol="telnet,tcp"
+
+			config_get_bool boolval "$cfg" remctl 0
+			[ "$boolval" -eq 1 ] && protocol="telnet(rfc2217),tcp"
+			;;
+		off)
+			enabled=0
+			;;
 		*) return 1
 	esac
-	config_get_bool boolval "$cfg" remctl 0
-	[ "$boolval" -gt 0 ] && $protocol="telnet(rfc2217),tcp,"
-
-	config_get timeout "$cfg" timeout 0
-	config_get device "$cfg" device
-	[ -z "$device" ] && return 1
 
 	config_get baudrate "$cfg" baudrate
-	config_get parity "$cfg" parity
-	parity=`echo "$parity" | head -c 1`
-	[ -z "$parity" ] && parity=n
+	if [ -n "$baudrate" ]; then
+		config_get parity "$cfg" parity
+		case "$parity" in
+			[Nn]one) parity=n ;;
+			[Oo]dd) parity=o ;;
+			[Ee]ven) parity=e ;;
+			"") ;;
+			*) return 1
+		esac
 
-	config_get databits "$cfg" databits
-	if [ -n "$databits" ]; then
-		[ "$databits" -lt 5 -o "$databits" -gt 8 ] && return 1
+		if [ -n "$parity" ]; then
+			config_get databits "$cfg" databits 8
+			[ "$databits" -ge 5 ] && [ "$databits" -le 9 ] || return 1
+
+			config_get stopbits "$cfg" stopbits 1
+			case "$stopbits" in
+				1) ;;
+				2) ;;
+				*) return 1
+			esac
+		fi
 	fi
 
-	config_get stopbits "$cfg" stopbits
-	case "$stopbits" in
-		1) ;;
-		2) ;;
-		"") ;;
-		*) return 1
-	esac
+	config_get timeout "$cfg" timeout 0
 
 	for key in rtscts local xonxoff nobreak hangup_when_done; do
 		config_get_bool boolval "$cfg" "$key"
 		[ -n "$boolval" ] || continue
 		key=`echo "$key" | tr '_' '-'`
-		options="$options,$key"
+		options="${options:+$options,}$key"
 		[ "$boolval" -eq 0 ] && options="$options=false"
 	done
 
-	config_list_foreach "$cfg" options list_cb_append options
+	config_list_foreach "$cfg" custom_options list_cb_append options
 
 	echo "connection: &$cfg"
 	echo "  accepter: $protocol,$port"
 	echo "  timeout: $timeout"
 	[ "$enabled" -eq 0 ] && echo "  enable: off"
-	echo "  connector: serialdev,$device,$baudrate$parity$databits$stopbits$options"
-	echo "  options:"
+	echo "  connector: serialdev,$device${baudrate:+,$baudrate${parity:+$parity$databits$stopbits}}${options:+,$options}${custom_options:+,$custom_options}"
 
 	for key in led_tx led_rx; do
 		config_get val "$cfg" "$key"
 		[ -n "$val" ] || continue
+		[ "$echo_options" -eq 1 ] && echo "  options:" && echo_options=0
 		key=`echo "$key" | tr '_' '-'`
-		echo "    $key: $val"
+		echo "    $key: *$val"
```

Comment:
Is this correct? the * refers to an anchor per yaml spec. (Examples under `define:` key in [stock ser2net.yaml](https://github.com/cminyard/ser2net/blob/master/ser2net.yaml))

Reply:
Yes, tested it. I used a Raspi with two GPIO LEDs. The UCI config looks like:
```
config proxy
         ...
        option led_tx 'tx'
        option led_rx 'rx'

config led 'tx'
        ...

config led 'rx'
        ...
```
Works for me(tm). :smile: